From b197f12d66ac844b72eb72ed152d73ddd7054983 Mon Sep 17 00:00:00 2001
From: Chris <meischris39@hotmail.com>
Date: Sun, 11 Oct 2026 16:28:26 +0100
Subject: [PATCH] [pictures] Don't change the display mode for a slideshow
 video

A video in a slideshow was opened like a film, so with "Adjust display
refresh rate" on, Kodi switched the display to the video's refresh rate
and the screen blanked as each one started. The video plays inside the
slideshow, so open it windowed.
---
 xbmc/pictures/GUIWindowSlideShow.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/xbmc/pictures/GUIWindowSlideShow.cpp b/xbmc/pictures/GUIWindowSlideShow.cpp
index e8bd98b..cac7031 100644
--- a/xbmc/pictures/GUIWindowSlideShow.cpp
+++ b/xbmc/pictures/GUIWindowSlideShow.cpp
@@ -37,6 +37,7 @@
 #include "resources/LocalizeStrings.h"
 #include "resources/ResourcesComponent.h"
 #include "settings/DisplaySettings.h"
+#include "settings/MediaSettings.h"
 #include "settings/Settings.h"
 #include "settings/SettingsComponent.h"
 #include "utils/Random.h"
@@ -1215,7 +1216,11 @@ bool CGUIWindowSlideShow::PlayVideo()
   CLog::Log(LOGDEBUG, "Playing current video slide {}", item->GetPath());
   m_bPlayingVideo = true;
   m_iVideoSlide = m_iCurrentSlide;
+  // The clip plays inside the slideshow, so it mustn't switch the display mode like a film
+  const bool startWindowed = CMediaSettings::GetInstance().DoesMediaStartWindowed();
+  CMediaSettings::GetInstance().SetMediaStartWindowed(true);
   bool ret = g_application.PlayFile(*item, "");
+  CMediaSettings::GetInstance().SetMediaStartWindowed(startWindowed);
   if (ret == true)
     return true;
   else
-- 
2.53.0

